### ISSUE TITLE
Package request: Private Internet Access VPN

### ISSUE THREAD

**Issue by StrongSand94191** [NixOS/nixpkgs#347539]
**Project description**

A private VPN

**Metadata**

* homepage URL: https://www.privateinternetaccess.com/
* source URL: https://github.com/pia-foss/desktop
* license: GPL3
* platforms: Windows, MacOS, Linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by bjornfor** [comment]
I guess this request is about the GUI, but FYI I wrote a PIA + NetworkManager NixOS module some time ago, and I configure my system like this:

```nix
{ config, lib, pkgs, ... }:

{
  networking.networkmanager.pia-vpn.enable = true;
  networking.networkmanager.pia-vpn.usernameFile = config.sops.secrets."pia_vpn_username".path;
  networking.networkmanager.pia-vpn.passwordFile = config.sops.secrets."pia_vpn_password".path;
  networking.networkmanager.pia-vpn.serverList =
    [ "denmark" "fi" "nl-amsterdam" "no" "sweden" "uk-london" "us-newyorkcity" ];

  sops.secrets."pia_vpn_username" = {};
  sops.secrets."pia_vpn_password" = {};
}
```

This allows enabling/disabling VPN with NetworkManager. Is there interest in having this in NixOS?

**Comment by StrongSand94191** [comment]
> I guess this request is about the GUI, but FYI I wrote a PIA + NetworkManager NixOS module some time ago, and I configure my system like this:
> 
> ```nix
> { config, lib, pkgs, ... }:
> 
> {
>   networking.networkmanager.pia-vpn.enable = true;
>   networking.networkmanager.pia-vpn.usernameFile = config.sops.secrets."pia_vpn_username".path;
>   networking.networkmanager.pia-vpn.passwordFile = config.sops.secrets."pia_vpn_password".path;
>   networking.networkmanager.pia-vpn.serverList =
>     [ "denmark" "fi" "nl-amsterdam" "no" "sweden" "uk-london" "us-newyorkcity" ];
> 
>   sops.secrets."pia_vpn_username" = {};
>   sops.secrets."pia_vpn_password" = {};
> }
> ```
> 
> This allows enabling/disabling VPN with NetworkManager. Is there interest in having this in NixOS?

yes that would be cool

**Comment by auroraveon** [comment]
i haven't read your code yet but doesn't this put the username and password in the nix store?

**Comment by bjornfor** [comment]
> i haven't read your code yet but doesn't this put the username and password in the nix store?

No, that info has to be passed in `usernameFile` and `passwordFile`, exactly to avoid that.
